### PR TITLE
Easier to use static tool panel views.

### PR DESCRIPTION
**Pull request by jmchilton**
I'm about to publish "a thing" for managing tool panel views by utilizing data about tools from many sources and the process of thinking through how to apply these across multiple servers and in multiple contexts has made a couple initial shortcomings for a very obvious use case clear. This is pull request fixes those up and includes tests.

Two related use cases - name "I have a list of deprecated tools I want to hide from selected sections" and "I have a list of deprecated tools I just want to apply to the original tool panel" - are both now pretty and natural and were pretty awkward before.

The first commit allows the list of exclusion criteria applied to the root of the panel apply to sections as well. This allows effectively global exclusions of tools - by ID, regex, etc.. Previously if you wanted to exclude say TopHat - you'd need to know a priori what each section it is in and apply the filter there. Sections may have different names, IDs, etc.. across servers so this can be hard to apply in a uniform fashion across multiple servers.

The second commit allows the root panel view to just not contain a set of items to describe this view. In this case the default is a description of the integrated tool panel. Combined with the first change - this allows very simple filters that just exclude tools and apply to any tool panel.

Ideally we could get this merged before 21.09 😇🙏 .

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/tool_util/toolbox/views/definitions.py [hunk 1]
```diff
@@ -108,7 +112,7 @@ class Section(BaseModel, HasItems):
     content_type: Literal['section'] = Field(alias="type")
     id: Optional[str]
     name: Optional[str]
-    items: Optional[List[SectionContent]]
+    items: Optional[List[RootContent]]  # really is just SectionContent but would need to use type variables to represent that.
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/toolbox/views/definitions.py]:
```suggestion
    items: Optional[List['RootContent']]  # really is just SectionContent but would need to use type variables to represent that.
```
?

**Comment by mvdbeek** [comment]
The unit test errors seem relevant:
```
E   pydantic.errors.ConfigError: field "items" not yet prepared so type is still a ForwardRef, you might need to call Section.update_forward_refs().

```

File: lib/galaxy/tool_util/toolbox/views/definitions.py [hunk 2]
```diff
@@ -77,8 +77,11 @@ class ItemsFrom(BaseModel):
 ]
 
 
-class HasItems:
-    items: Optional[List['RootContent']]
+C = TypeVar('C', bound="RootContent")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/toolbox/views/definitions.py]: Would this be an alternative? https://pydantic-docs.helpmanual.io/usage/postponed_annotations/#self-referencing-models
I guess that's what the error message was hinting at ?

Comment (jmchilton) [reply lib/galaxy/tool_util/toolbox/views/definitions.py]: I had a version that used that but Section allowing all RootContent allows Sections and this broke a unit test and is just wrong.

This seemed to be better on Python 3.8... but I guess it fails under Python 3.6... hmm... sorry if this is slowing down the release.

Comment (mvdbeek) [reply lib/galaxy/tool_util/toolbox/views/definitions.py]: No worries about the release, we moved the freeze to Thursday anyway, lot of open PRs at this point.

Comment (jmchilton) [reply lib/galaxy/tool_util/toolbox/views/definitions.py]: xref https://github.com/samuelcolvin/pydantic/issues/1547#issuecomment-636466257

Well at least I'm certain "the correct" solution won't work so I can pick a hack and feel okay.

**Comment by mvdbeek** [comment]
```
galaxy.web.framework.decorators ERROR 2021-09-15 16:36:53,119 [pN:main,p:3215,tN:ThreadPoolExecutor-24_7] Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/web/framework/decorators.py", line 312, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/webapps/galaxy/api/jobs.py", line 236, in show
    return view_show_job(trans, job, full_output)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/managers/jobs.py", line 421, in view_show_job
    job_dict.update(tool_name=tool.name)
AttributeError: 'NoneType' object has no attribute 'name'
```
That's an unrelated error, I can fix this

**Pull request merged**